targets/TARGET_STM/TARGET_STM32WB/TARGET_STM32WB55xG/device/TOOLCHAIN_ARM_MICRO/stm32wb55xx.sct

```diff
@@ -0,0 +1,73 @@
+#! armcc -E
+; Scatter-Loading Description File
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Copyright (c) 2015, STMicroelectronics
+; All rights reserved.
+;
+; Redistribution and use in source and binary forms, with or without
+; modification, are permitted provided that the following conditions are met:
+;
+; 1. Redistributions of source code must retain the above copyright notice,
+;     this list of conditions and the following disclaimer.
+; 2. Redistributions in binary form must reproduce the above copyright notice,
+;    this list of conditions and the following disclaimer in the documentation
+;    and/or other materials provided with the distribution.
+; 3. Neither the name of STMicroelectronics nor the names of its contributors
+;    may be used to endorse or promote products derived from this software
+;    without specific prior written permission.
+;
+; THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+; AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+; IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+; DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+; FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+; DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+; SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+; CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+; OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+; OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+#if !defined(MBED_APP_START)
+#define MBED_APP_START 0x08000000
+#endif
+
+#if !defined(MBED_APP_SIZE)
+; 512KB FLASH
+#define MBED_APP_SIZE 0x80000
+#endif
+
+#if !defined(MBED_BOOT_STACK_SIZE)
+  #define MBED_BOOT_STACK_SIZE 0x400
+#endif
+
+#define Stack_Size MBED_BOOT_STACK_SIZE
+
+; 512KB FLASH (0x80000) + 192KB SRAM (0x30000) +  Shared mem
+LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
+
+  ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address
+   *.o (RESET, +First)
+   *(InRoot$$Sections)
+   .ANY (+RO)
+  }
+
+  ; Total: 79 vectors = 316 bytes (0x13C) to be reserved in RAM
+  RW_IRAM1 (0x20000000+0x13C) (0x30000-0x13C-Stack_Size)  {  ; RW data
+  .ANY (+RW +ZI)
+  }
+
+  ARM_LIB_STACK (0x20000000+0x30000) EMPTY -Stack_Size { ; stack
+  }
+
+  ; SRAM2 - Shared memory
+  RW_IRAM2a 0x20030000 0x00002800  {  ; RW data
+#if defined(BLE)
+   *(MAPPING_TABLE, +First)
```

Comment:
is the order important here (thus using FIRST attribute) ? I can see its used in IAR as well not for GCC ARM - I do not see there any preprocessor for this section as here (if defined BLE) ?

I suspect you are getting FIRST/LAST error because MAPPING_TABLE is not defined in your code. If BLE is not present, should at least be defined ?

Reply:
@0xc0170  The #if defined(BLE) actually does not work: the condition is always considered FALSE. This results in MBED2 compiling ok but has created a regression on mbed-os BLE use cases, That's why target_cordio-tests-* are failed now with ARM TC.  

If I remove this condition, cordio tests are ok, but MBED2 link will fail. 

Since then I tried replacing BLE by FEATURE_BLE or other macros names, but still doesn't work. I can't find a way to define a condition related to BLE or mbed-os vs. MBED2 to solve my issue ... that's why I'd like guidance.

And yes the +First attribute is needed as the mapping table needs to be mapped at this exact place. Without FIRST I understand that ARM linker will not ensure the order. I think other linker ensure the order without a specific attribute.  If you think that  my understanding is wrong, then I would welcome any other proposal to test ...

Reply:
The order is specified here http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0474l/Chunk364384667.html

Where is MAPPING_TABLE defined?  I cant find it. In the startup you define only these

```
; Vector Table Mapped to Address 0 at Reset
                AREA    RESET, DATA, READONLY
```

Reply:
`FEATURE_BLE` should be defined if this feature is enabled

Reply:
FEATURE_BLE is defined for all C files (BTW it is also defined in MBED2 application even if the BLE feature is not compiled) but it does not seem to be taken into account for linker script ...

If I look at the files .profile-ld it seems to contain several list of macros: 'flags' list, 'macros' list and 'symbols' list.  Like MBED_BOOT_STACK_SIZE which is used as well in linker script, is part of the 'flags' list, not in the macros, nor in the symbols ... 
```
    "flags": [
        "--keep=os_cb_sections", 
        "--predefine=\"-DMBED_BOOT_STACK_SIZE=4096\"", 
        "--show_full_path"
    ], 
```
So I am wondering how to define a 'flag' that will be defined in mbed-os when BLE feature is compiled and not defined otherwise.

Reply:
> The order is specified here http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0474l/Chunk364384667.html

Which chapter are you pointing to ?

Reply:
Wrong copy paste, Default section placement is the chapter .

Taken from there:

```
By default, the linker places input sections in the following order within an execution region:

By attribute as follows:

Read-only code.

Read-only data.

Read-write code.

Read-write data.

Zero-initialized data.

By input section name if they have the same attributes. Names are considered to be case-sensitive and are compared in alphabetical order using the ASCII collation sequence for characters.

By a tie-breaker if they have the same attributes and section names. By default, it is the order that armlink processes the section. You can override this with the FIRST or LAST execution region attribute.
```

Reply:
hmm interesting - that means that alphabetical order should be enough for taking care of my issue here ...

Reply:
That works fine - thanks @0xc0170  
Much simpler and straightforward than what I was searching for :-) 
I have updated the PR with the change !

Reply:
Awesome ! So after all how it was fixed? To know in the future if we have this problem again :-)

Reply:
I have added an EDIT in previous comment: 
**EDIT: SOLVED with Martin support:** no need to use the +First attribute as the alphabetical order will ensure that MAPPING_TABLE is placed before MB_MEM1 : tested OK and branch updated.